### PR TITLE
Fixes #12277. Enum creation form.

### LINKED ISSUE #12277: VIsual enum editor adds extra (blank) values
When using the visual enum editor to add less than four enum values, it will add blank values. For instance, `` `e_test` enum('a','b','','') NOT NULL``

### PR DESCRIPTION
**Pull request by kkarol012**
Fixes #12277

### CODE REVIEW

File: js/functions.js [hunk 1]
```diff
@@ -3424,14 +3424,24 @@ AJAX.registerOnload('functions.js', function () {
...
         var dropIcon = Functions.getImage('b_drop');
         for (i = 0; i < values.length; i++) {
+            if (values[i] == '' && isEditing) {
+                checkedString = 'checked';
+            } else {
+                checkedString = '';
+            }
             fields += '<tr><td>' +
+                   '<div class=\'input_container\'>' +
+                   '<input type=\'checkbox\'' + checkedString + '>empty string' +
```
Comment (williamdes) [inline js/functions.js]: can you please use PMA_Messages like it is done in the codebase ?
`js/messages.php`

Comment (kkarol012) [reply js/functions.js]: done

**Review by williamdes — APPROVED** [review]
Code looks okay, need to test :)

File: js/functions.js [hunk 2]
```diff
@@ -3424,14 +3424,25 @@ AJAX.registerOnload('functions.js', function () {
...
+        var checkedString;
         for (i = 0; i < values.length; i++) {
+            if (values[i] === '' && isEditing) {
+                checkedString = 'checked';
+            } else {
+                checkedString = '';
+            }
             fields += '<tr><td>' +
+                   '<div class=\'input_container\'>' +
+                   '<input type=\'checkbox\'' + checkedString + '>' + Messages.enum_emptyString +
```
Comment (mauriciofauth) [inline js/functions.js]:
```suggestion
                   '<input type="checkbox"' + checkedString + '>' + Messages.enum_emptyString +
```

File: js/functions.js [hunk 3]
```diff
@@ -3424,14 +3424,25 @@ AJAX.registerOnload('functions.js', function () {
...
         if (values.length === 0) {
             values.push('', '', '', '');
+            isEditing = false;
         }
         // Add the parsed values to the editor
         var dropIcon = Functions.getImage('b_drop');
+        var checkedString;
         for (i = 0; i < values.length; i++) {
+            if (values[i] === '' && isEditing) {
+                checkedString = 'checked';
```
Comment (mauriciofauth) [inline js/functions.js]:
```suggestion
                checkedString = ' checked';
```

File: js/functions.js [hunk 4]
```diff
@@ -3663,7 +3677,10 @@ AJAX.registerOnload('functions.js', function () {
             $enumEditorDialog.find('.values')
                 .append(
                     '<tr class=\'hide\'><td>' +
+                    '<div class=\'input_container\'>' +
```
Comment (mauriciofauth) [inline js/functions.js]:
```suggestion
                    '<div class="input_container">' +
```

File: js/functions.js [hunk 5]
```diff
@@ -3663,7 +3677,10 @@ AJAX.registerOnload('functions.js', function () {
...
+                    '<input type=\'checkbox\'>' + Messages.enum_emptyString +
```
Comment (mauriciofauth) [inline js/functions.js]:
```suggestion
                    '<input type="checkbox">' + Messages.enum_emptyString +
```

**Review by MauricioFauth — CHANGES_REQUESTED** [review]
Thanks for your contribution.

Could you add `label` tags to the checkbox labels?

File: js/functions.js [hunk 6]
```diff
@@ -3424,14 +3424,25 @@ AJAX.registerOnload('functions.js', function () {
...
+            isEditing = false;
         }
         // Add the parsed values to the editor
         var dropIcon = Functions.getImage('b_drop');
+        var checkedString;
         for (i = 0; i < values.length; i++) {
+            if (values[i] === '' && isEditing) {
+                checkedString = 'checked';
+            } else {
+                checkedString = '';
```
[the remark this answers is not in the document]
Comment (kkarol012) [reply js/functions.js]: changed

File: js/functions.js [hunk 7]
```diff
@@ -3424,14 +3424,25 @@ AJAX.registerOnload('functions.js', function () {
...
+            }
             fields += '<tr><td>' +
+                   '<div class=\'input_container\'>' +
```
[the remark this answers is not in the document]
Comment (kkarol012) [reply js/functions.js]: fixed

**Comment by williamdes** [comment]
I had a JS error while adding a value
```json
{
    "pma_version": "5.1.0-dev",
    "browser_name": "CHROME",
    "browser_version": "81.0.4044.113",
    "user_os": "Linux",
    "server_software": "nginx/1.17.8",
    "user_agent_string": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.113 Safari/537.36",
    "locale": "fr",
    "configuration_storage": "enabled",
    "php_version": "8.0.0-dev",
    "script_name": "index.php",
    "exception_type": "js",
    "exception": {
        "mode": "stack",
        "name": "TypeError",
        "message": "Cannot read property 'value' of undefined",
        "stack": [
            {
                "line": "292",
                "func": "",
                "context": [
                    "            if (lastException === ex) {",
                    "                processLastException();",
                    "            }",
                    "        }, (stack.incomplete ? 2000 : 0));",
                    "",
                    "        throw ex; // re-throw to propagate to the top level (and cause wind//...",
                    "    }",
                    "",
                    "    report.subscribe = subscribe;",
                    "    report.unsubscribe = unsubscribe;",
                    "    return report;"
                ],
                "column": "",
                "uri": "js/vendor/tracekit.js?v=5.1.0-dev",
                "scriptname": "js/vendor/tracekit.js"
            },
            {
                "func": "HTMLDivElement.",
                "line": "3537",
                "column": "37",
                "context": [
                    "            // When the submit button is clicked,",
                    "            // put the data back into the original form",
                    "            var valueArray = [];",
                    "            $(this).find('.input_container').each(function (index, elm) {",
                    "                var childs = $(elm).children('input');",
                    "                var val = childs[1].value.replace(/\\\\/g, '\\\\\\\\').replace(/'//...",
                    "                if (val !== '' || childs[0].checked) {",
                    "                    valueArray.push('\\'' + val + '\\'');",
                    "                }",
                    "            });",
                    "            // get the Length/Values text field where this value belongs"
                ],
                "uri": "js/functions.js?v=5.1.0-dev",
                "scriptname": "js/functions.js"
            },
            {
                "func": "Function.each",
                "line": "2",
                "column": "2976",
                "context": [
                    "/*! jQuery v3.5.0 | (c) JS Foundation and other contributors | jquery.org/l//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.1.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "jQuery.fn.init.each",
                "line": "2",
                "column": "1454",
                "context": [
                    "/*! jQuery v3.5.0 | (c) JS Foundation and other contributors | jquery.org/l//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.1.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "HTMLDivElement.buttonOptions.",
                "line": "3535",
                "column": "46",
                "context": [
                    "        var buttonOptions = {};",
                    "        buttonOptions[Messages.strGo] = function () {",
                    "            // When the submit button is clicked,",
                    "            // put the data back into the original form",
                    "            var valueArray = [];",
                    "            $(this).find('.input_container').each(function (index, elm) {",
                    "                var childs = $(elm).children('input');",
                    "                var val = childs[1].value.replace(/\\\\/g, '\\\\\\\\').replace(/'//...",
                    "                if (val !== '' || childs[0].checked) {",
                    "                    valueArray.push('\\'' + val + '\\'');",
                    "                }"
                ],
                "uri": "js/functions.js?v=5.1.0-dev",
                "scriptname": "js/functions.js"
            },
            {
                "func": "HTMLButtonElement.",
                "line": "11",
                "column": "13714",
                "context": [
                    "(function(t){\"function\"==typeof define&&define.amd?define([\"jquery\"],t):t(j//...",
                    "},_convertPositionTo:function(t,e){e||(e=this.position);var i=\"absolute\"===//...",
                    "t(n.alsoResize).each(function(){var e=t(this),s=t(this).data(\"ui-resizable-//...",
                    "},_destroyIcons:function(){this._removeClass(this.headers,\"ui-accordion-ico//...",
                    "},_updateLabel:function(){var t=this.label.contents().not(this.element[0]);//...",
                    "break;case U[1]-1:T+=\" ui-datepicker-group-last\",I=\" ui-corner-\"+(Y?\"left\"://...",
                    "},_setupEvents:function(){this._off(this.handles),this._on(this.handles,thi//...",
                    "}),o.closing=!0,this._trigger(\"close\",e,{tooltip:i}),o.hiding||(o.closing=!//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery-ui.min.js?v=5.1.0-dev",
                "scriptname": "js/vendor/jquery/jquery-ui.min.js"
            },
            {
                "func": "HTMLButtonElement.newFunc",
                "line": "265",
                "column": "33",
                "context": [
                    "     */",
                    "    wrapFunction: function (func) {",
                    "        if (!func.wrapped) {",
                    "            var newFunc = function () {",
                    "                try {",
                    "                    return func.apply(this, arguments);",
                    "                } catch (x) {",
                    "                    TraceKit.report(x);",
                    "                }",
                    "            };",
                    "            newFunc.wrapped = true;"
                ],
                "uri": "js/error_report.js?v=5.1.0-dev",
                "scriptname": "js/error_report.js"
            },
            {
                "func": "HTMLButtonElement.dispatch",
                "line": "2",
                "column": "43107",
                "context": [
                    "/*! jQuery v3.5.0 | (c) JS Foundation and other contributors | jquery.org/l//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.1.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            },
            {
                "func": "HTMLButtonElement.$event.dispatch",
                "line": "374",
                "column": "19",
                "context": [
                    "$event.dispatch = function( event ){",
                    "\tif ( $.data( this, \"suppress.\"+ event.type ) - new Date().getTime() > 0 ){",
                    "\t\t$.removeData( this, \"suppress.\"+ event.type );",
                    "\t\treturn;",
                    "\t}",
                    "\treturn $dispatch.apply( this, arguments );",
                    "};",
                    "",
                    "// event fix hooks for touch events...",
                    "var touchHooks = ",
                    "$event.fixHooks.touchstart = "
                ],
                "uri": "js/vendor/jquery/jquery.event.drag-2.2.js?v=5.1.0-dev",
                "scriptname": "js/vendor/jquery/jquery.event.drag-2.2.js"
            },
            {
                "func": "HTMLButtonElement.v.handle",
                "line": "2",
                "column": "41091",
                "context": [
                    "/*! jQuery v3.5.0 | (c) JS Foundation and other contributors | jquery.org/l//...",
                    "!function(e,t){\"use strict\";\"object\"==typeof module&&\"object\"==typeof modul//...",
                    ""
                ],
                "uri": "js/vendor/jquery/jquery.min.js?v=5.1.0-dev",
                "scriptname": "js/vendor/jquery/jquery.min.js"
            }
        ],
        "incomplete": "false",
        "partial": "true",
        "uri": "index.php?route=%2Ftable%2Fstructure"
    }
}
```

**Pull request closed without merging**